x/exomint/keeper/keeper.go

```diff
@@ -0,0 +1,76 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ExocoreNetwork/exocore/x/exomint/types"
+)
+
+type (
+	Keeper struct {
+		cdc              codec.BinaryCodec
+		storeKey         storetypes.StoreKey
+		bankKeeper       types.BankKeeper
+		epochsKeeper     types.EpochsKeeper
+		feeCollectorName string
+		// the address capable of executing a MsgUpdateParams message, typically x/gov.
+		authority string
+	}
+)
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeKey storetypes.StoreKey,
+	ak types.AccountKeeper,
+	bk types.BankKeeper,
+	ek types.EpochsKeeper,
+	feeCollectorName string,
+	authority string,
+) Keeper {
+	// ensure mint module account is set
+	if addr := ak.GetModuleAddress(types.ModuleName); addr == nil {
+		panic(fmt.Sprintf("the x/%s module account has not been set", types.ModuleName))
+	}
+
+	return Keeper{
+		cdc:              cdc,
+		storeKey:         storeKey,
+		bankKeeper:       bk,
+		epochsKeeper:     ek,
+		feeCollectorName: feeCollectorName,
+		authority:        authority,
+	}
+}
+
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+// MintCoins implements an alias call to the underlying supply keeper's
+// MintCoins to be used in BeginBlocker.
+func (k Keeper) MintCoins(ctx sdk.Context, newCoins sdk.Coins) error {
+	if newCoins.Empty() {
+		// skip as no coins need to be minted
```

Comment:
Would this happen? As if there's no coins need to be minted, will this function be called?

Reply:
It is a bit of an excessive short-circuit mechanism. I have removed it in the next commit.